### ISSUE TITLE
Fix bug on adding skills to new tasks

### ISSUE THREAD

**Issue by JoshSmith** [code-corps/code-corps-ember#1161]
# Problem

---

From @begedin:

While part of the behavior was fixed with the latest commit (and already rebased), the UI itself still has a bug, which will take a longer time to fix. 

If I add a skill using the project skill list, then try to search for the same skill in the dropdown, it will appear "uncrossed", as if not yet added, in the results:

![image](https://cloud.githubusercontent.com/assets/1925540/23950429/3f66538e-098b-11e7-8579-a090203bd3a5.png)

Even though it looks incorrectly, clicking it will still remove it, not add a duplicate, so at least the background logic works.

Basically,
- `skill-typeahead-result` usually uses `service:x-skills.includes(skill)` to determine if skill has already been added. this method is NOT easily rewritten into something else
- In the case of `task.new`, the `skill-typeahead-result` instead uses just an `array.includes(skill)`, but this no longer works when we add project skills into the mix, because those are `DS.PromiseObject` records, not `DS.Model` records. This happened to work because the method was named the same, but the proper method to use here is `array.isAny('id', skillId)` so it works with both `DS.Model` and `DS.PromiseObject` records

- monkeypatch the `includes` method in this specific case - quick and dirty, definitely against it
- wrap the unsaved skills collection into another service, which makes it work - clean, but makes thing worse in the long run, I think. Another complex service, difficult to follow.
- rethink the `projectSkillsList` component added here. make it an extremely simple `skills-item-list` component which just receives and renders a list of skills. let the controller handle the exclusion of already added skills, etc. - clean and doesn't make things worse; a good compromise.
- rewrite the services, rethink them, really. now that I got back into them, I found them extremely difficult to follow and, especially with "sometimes" accessing the `content` property of proxy objects, ex: 

```Javascript
return records.any((found) => {
  let targetId = get(target, 'id');
  let targetModelName = 
     get(target, 'constructor.modelName') || get(target, 'content.constructor.modelName');
  let foundId = found.belongsTo(targetModelName).id();

  return (foundId === targetId);
});
```

I think, with proper controller usage, the whole xSkillsList collection of services could be almost if not completely eliminated.

**Comment by begedin** [comment]
@JoshSmith The bug visible in the first give here should not be happening anymore. That part was supposed to be solved. Did we have a regression?

The only problem now is, if you add a skill via the project skill list, then search for it in the dropdown, the search result will look like the skill has not been added. Clicking the search result will still remove the skill, not add a duplicate like the gif is saying.

**Comment by JoshSmith** [comment]
@begedin you're right, I think. Please feel free to update the description accordingly.

Also think we need some more discussion here about what we're trying to accomplish here. I think this would be a good synchronous conversation since it's a big thing we're discussing.

**Comment by daveconnis** [comment]
Has there been any more discussion around this?

**Comment by begedin** [comment]
@daveconnis Not really. The same issue is still present and the lists are pretty convoluted. It's fixable, but a fix would make it more convoluted. The suggested rewrite would require some work, but would make it easier to manage, probably.
